addon/components/paper-item.js

```diff
@@ -49,6 +52,26 @@ export default Component.extend(ParentMixin, {
     return proxiedComponents.objectAt(0);
   }),
 
+  didInsertElement() {
+    this._super(...arguments);
+
+    this._mouseEnterHandler = this.handleMouseEnter.bind(this);
+    this._mouseLeaveHandler = this.handleMouseLeave.bind(this);
+
+    this.element.addEventListener('mouseenter', this._mouseEnterHandler);
+    this.element.addEventListener('mouseleave', this._mouseLeaveHandler);
+  },
+
+  willDestroyElement() {
+    this._super(...arguments);
+
+    this.element.removeEventListener('mouseenter', this._mouseEnterHandler);
+    this.element.removeEventListener('mouseleave', this._mouseLeaveHandler);
+
+    this._mouseEnterHandler = undefined;
+    this._mouseLeaveHandler = undefined;
```

Comment:
@fpauser Why do we have to define a handler property to store the action and then clear it in will destroy ? Instead we can directly assign the actions to event listener and remove it in will destroy hook.

Reply:
Its needed because `bind()` returns a new function bound to `this` with each call. To have `removeEventListener` remove our handler we have to give it the exact same handler function that was given to `addEventHandler`.

```js
this.handleMouseEnter.bind(this) !== this.handleMouseEnter.bind(this)
```

Reply:
Otherwise we'd produce leaks.

Reply:
That is a good insight. Thanks